docs/dev/stream/state/state.md

```diff
@@ -266,6 +266,101 @@ a `ValueState`. Once the count reaches 2 it will emit the average and clear the
 we start over from `0`. Note that this would keep a different state value for each different input
 key if we had tuples with different values in the first field.
 
+### State time-to-live (TTL)
+
+A time-to-live (TTL) can be assigned to the keyed state value. 
+In this case it will expire after the configured TTL
+and its stored value will be cleaned up on the best effort basis which is discussed in details later.
+
+The state collection types support per-entry TTLs: list elements and map entries expire independently.
+
+To use state TTL you must first build a `StateTtlConfiguration` object:
+
+<div class="codetabs" markdown="1">
```

Comment:
Show in the first example how this feature is used. Not only the configuration without context. Maybe with a example state for which the config is set.